File: 03_palindrom.rs
fn main() {    
    let mut input_line = String::new();
    std::io::stdin()
    .read_line(&mut input_line)
    .expect("Failed to read line");

    let reverse = &input_line.to_lowercase().trim().chars().rev().collect::<String>();
    println!("{}", input_line.to_lowercase().trim() == reverse);
}

File: 03 Palindrom.ts
import readline from 'readline';

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
    terminal: false
});

const jePalindrom = (line) => String(line)
.toLowerCase()
.split("")
.every((el, index, list) => el == list.at(list.length - 1 - index));

rl.on('line', line => {
    console.log(jePalindrom(line) ? "ano" : "ne");
})